Version 2.5.4 (Authentication Firebase, reports, books, live broadcast, and push)

pablogrela opened the pull request:
Authentication Firebase
Manage reports
Manage reserves
Push notifications
Another improves and functionalities

pablogrela committed:
Se ha refactoriazado el codigo de home, también se quitaron los métodos deprecated.
Se ha separada la url base, a una de programmes y otra de station
Se ha cambiado la url de los programas a https://cuacfm.org/radioco/programmes/json/
Se ha cambiado el textView de no elements para que tuviese una mejor apariencia

pablogrela committed:
Se ha refactoriazado el codigo de home, también se quitaron los métodos deprecated.
Se ha separada la url base, a una de programmes y otra de station
Se ha cambiado la url de los programas a https://cuacfm.org/radioco/programmes/json/
Se ha cambiado el textView de no elements para que tuviese una mejor apariencia

pablogrela committed:
Se cambio la url de los programas de cuacfm a:
api/1/radiocom/programmes
Siguiendo las indicaciones del creador de radioco.

pablogrela committed:
Se ha vuelto a poner una url en vez de dos, por petición del creador, despues se vifurca en programmes y radiostation

pablogrela committed:
Now the information of twitter and facebook is obtained from the StationDTO, which receives the information of the url provided.
If no information is provided on the StationDTO, the information provided in the GlobalValues is collected.

pablogrela committed:
Added the position of the radio station.
The length and latitude are taken from the StationDTO.

pablogrela committed:
Added the position of the radio station.
The length and latitude are taken from the StationDTO.

pablogrela committed:
-Rename the following methods like this:
chargueEmisora->loadStation
chargueGaleria->loadGallery
chargueEmsision->loadNews
charguePodcast->loadPodcast
chargueMap->loadMap
chargueTwitter->loadTwitter
chargueFacebook->loadFacebook
ProcessBuilder->processBuilder
chargueDeveloper->loadDeveloperInfo

In loadMap, has put the two types of intent, one to put a mark on the map and oto that sets the route from the source where the user.

pablogrela committed:
Update to version 1.0.3

pablogrela committed:
Update to version 1.0.3

ficiverson committed:
Hotfix (#4)

* v1.0

- Ready to go live on cuac fm

* v1.0

- Pre Release CUAC FM
- Add support to search podcast info to filter list

* v1.0

- Hotfix search on podcast
- UI helper on initial tutorial to show user how to scroll down
- Added clases from discrollable view to the project to avoid problems with manifest version on android API

pablogrela committed:
Update to version 1.0.4

pablogrela committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	app/src/main/java/justforcommunity/radiocom/activities/Home.java
#	app/src/main/java/justforcommunity/radiocom/fragments/NewsPageFragment.java
#	app/src/main/java/justforcommunity/radiocom/model/StationDTO.java
#	app/src/main/java/justforcommunity/radiocom/service/ServiceGetPrograms.java
#	app/src/main/java/justforcommunity/radiocom/utils/GlobalValues.java
#	app/src/main/res/values/strings.xml
#	build.gradle

pablogrela committed:
Added firebase authentication, with email/password authenticate and google sign in.
Also it added verify email, reset password and url to signup in members.

pablogrela committed:
Version 2.2.0
Authenticate Firebase
Manage Incidences with Members
 - View incidence list
 - Report new incidence

pablogrela committed:
Version 2.2.1
Manage Incidences with Members
 - Added management and sending photos of incidences

pablogrela committed:
Version 2.2.2
Change name incidence to report
Another improves in reports

pablogrela committed:
Version 2.2.1
Corrected errors

pablogrela committed:
Version 2.2.2
Member account management
   - Permits for restricted views

pablogrela committed:
Version 2.2.2
Changes and fixes

pablogrela committed:
Version 2.2.2
Changes

pablogrela committed:
Version 2.2.3
Member account management
   - Done incomplete task of reports
   - Improved appearance
   - Add messaging Firebase
   - Done format all classes
   - Another changes

pablogrela committed:
Version 2.3.0
   Added manage reserves
   Redirect with authentication to members
   Another fix and bugs

pablogrela committed:
Version 2.3.1
Changes

pablogrela committed:
Version 2.3.2
Fixes and improves

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/CreateReserve.java:
```diff
@@ -0,0 +1,307 @@
+import static justforcommunity.radiocom.utils.GlobalValues.REST_URL;
+
+
+public class CreateReserve extends AppCompatActivity {
```
it makes more sense to rename to CreateBook

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/CreateReport.java:
```diff
@@ -0,0 +1,448 @@
+    }
+
+    // Create new report with form
+    private void createReport() {
```
this method is really difficult to understand

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Gallery.java:
```diff
@@ -25,33 +25,26 @@
- * Created by iver on 5/9/16.
- */
-public class Gallery extends AppCompatActivity{
+public class Gallery extends AppCompatActivity {
```
why did you delete the created by?

pablogrela replied:
it is important? 
I was only in some classes that you created, not at all, so it is homogeneous.

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Home.java:
```diff
@@ -167,13 +189,14 @@ public void onClick(View v) {
     public void onStart() {
         super.onStart();
         GoogleAnalytics.getInstance(this).reportActivityStart(this);
+        //updateUserInfo();
```
delete this line if its not necessary

ficiverson replied:
This line remains

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Home.java:
```diff
@@ -344,6 +378,21 @@ public boolean onNavigationItemSelected(MenuItem item) {
+                loadReserve();
+                break;
+            case R.id.nav_members:
+                processBuilder(mContext, this, GlobalValues.membersAPI + "?token=" + token);
```
put any string param as a final static String on GlobalValues

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Home.java:
```diff
@@ -407,6 +470,34 @@ public void loadPodcast() {
+        processFragment(reportPageFragment, mContext.getString(R.string.action_report));
+    }
+
+    public void loadUserReserve() {
```
rename Reserve to Book

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Home.java:
```diff
@@ -497,21 +587,60 @@ private void audioController() {
-        CustomTabsIntent customTabsIntent = builder.build();
-        customTabsIntent.launchUrl(this, Uri.parse(url));
+    public void updateUserInfo() {
+
```
remove this comment code

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Report.java:
```diff
@@ -0,0 +1,292 @@
+            photos_head.setVisibility(View.VISIBLE);
+        }
+        for (String nameFile : report.getFiles()) {
+            String url = imageReportURL + report.getId() + "?imageName=" + nameFile + "&token=" + token;
```
change this params with a static final String

pablogrela replied:
Difficult
report.getId() is dynamic, and name file

ficiverson replied:
i was talking about "?imageName=" and "&token="

pablogrela replied:
done

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Reserve.java:
```diff
@@ -0,0 +1,257 @@
+import static justforcommunity.radiocom.utils.GlobalValues.MANAGE;
+import static justforcommunity.radiocom.utils.GlobalValues.RESERVE_JSON;
+
+public class Reserve extends AppCompatActivity {
```
rename to book

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Reserve.java:
```diff
@@ -0,0 +1,257 @@
+                if (!intent.getBooleanExtra("notificationSkip", false)) {
+                    Intent i = new Intent(Reserve.this, PodcastingService.class);
+                    stopService(i);
+                    //notifyAdapterToRefresh();
```
remove this code

ficiverson commented on app/src/main/java/justforcommunity/radiocom/adapters/ReportListAdapter.java:
```diff
@@ -0,0 +1,161 @@
+
+    private class ItemFilter extends Filter {
+        @Override
+        protected FilterResults performFiltering(CharSequence constraint) {
```
is filter enable on UI if not please remove this code

pablogrela replied:
yes, filter is enable

ficiverson replied:
perfect!!

ficiverson commented on app/src/main/java/justforcommunity/radiocom/adapters/ReserveListAdapter.java:
```diff
@@ -0,0 +1,158 @@
+        return mFilter;
+    }
+
+    private class ItemFilter extends Filter {
```
is filter enabled? if its not enabled please remove this code

pablogrela replied:
yes, filter is enable

ficiverson commented on app/src/main/java/justforcommunity/radiocom/fragments/ReserveUserPageFragment.java:
```diff
@@ -0,0 +1,88 @@
+import static justforcommunity.radiocom.utils.GlobalValues.reservesUserURL;
+
+
+public class ReserveUserPageFragment extends ReservePageFragment {
```
Rename Reserve

ficiverson commented on app/src/main/java/justforcommunity/radiocom/model/ReserveDTO.java:
```diff
@@ -0,0 +1,135 @@
+
+import java.util.Date;
+
+public class ReserveDTO {
```
rename reserve

ficiverson commented on app/src/main/java/justforcommunity/radiocom/service/ServiceAccounts.java:
```diff
@@ -0,0 +1,70 @@
+
+    public AccountDTO getAccount() throws RestClientException, WebServiceStatusFailException {
+
+        String url = accountURL + "?token=" + getTokenFirebase() + "&deviceToken=" + FirebaseInstanceId.getInstance().getToken();
```
please let the urls on GlobalValues

pablogrela replied:
BookDTO?

ficiverson replied:
ok

ficiverson commented on app/src/main/java/justforcommunity/radiocom/service/ServiceElements.java:
```diff
@@ -0,0 +1,77 @@
+
+    public List<ElementDTO> getElements(String restURL) throws RestClientException, WebServiceStatusFailException {
+
+        String url = restURL + "?token=" + getTokenFirebase();
```
GlobalValues

ficiverson commented on app/src/main/java/justforcommunity/radiocom/service/ServiceReports.java:
```diff
@@ -0,0 +1,148 @@
+        ResponseEntity<String> response;
+
+        try {
+            agregarCabeceras(getRequestHeaders());
```
rename this method to english

ficiverson commented on app/src/main/java/justforcommunity/radiocom/task/FirebaseUtils.java:
```diff
@@ -0,0 +1,88 @@
+        if (result) {
+            firebaseActivity.setToken(token);
+        } else {
+            //mActivity.resultKO();
```
delete this else statement

ficiverson commented on app/src/main/java/justforcommunity/radiocom/task/Report/GetReports.java:
```diff
@@ -0,0 +1,86 @@
+                res = true;
+
+            } catch (RestClientException e) {
+                Log.e(TAG, "doInBackground()", e);
```
remove useless logs

pablogrela replied:
I do not get it¡
Is it an error log, I put it in debug mode?

ficiverson replied:
yes please use debug mode

pablogrela replied:
done, I changed all Log.e to Log.d

ficiverson commented on app/src/main/java/justforcommunity/radiocom/task/Reserve/Reserve/SendAnswerReserve.java:
```diff
@@ -0,0 +1,86 @@
+ *
+ */
+
+package justforcommunity.radiocom.task.Reserve.Reserve;
```
why the package is Reserve.Reserer?

pablogrela replied:
fail

ficiverson commented on app/src/main/java/justforcommunity/radiocom/utils/MessagingService.java:
```diff
@@ -0,0 +1,36 @@
+
+        // TODO(developer): Handle FCM messages here.
+        // Not getting messages here? See why this may be: https://goo.gl/39bRNJ
+        Log.d(TAG, "From: " + remoteMessage.getFrom());
```
Remove all verbose information on log

ficiverson commented on app/src/main/res/values/strings.xml:
```diff
@@ -53,6 +53,74 @@
     <string name="podcast_info_title">Descripción del programa:</string>
 
+    <!--Reports-->
+    <string name="account">Account</string>
```
cuenta

ficiverson commented on app/src/main/res/values/colors.xml:
```diff
@@ -24,6 +24,7 @@
     <color name="colorPrimary">#ff9900</color>
```
please put colorBlack85 back

ficiverson commented on app/src/main/res/values-v21/styles.xml:
```diff
@@ -25,5 +25,7 @@
         <item name="windowNoTitle">true</item>
         <item name="android:windowDrawsSystemBarBackgrounds">true</item>
         <item name="android:statusBarColor">@android:color/transparent</item>
+        <item name="android:textAllCaps">false</item>
```
be carefull your are applying this global

pablogrela replied:
I delete it?

ficiverson replied:
no its just a reminder

ficiverson reviewed (commented):
There are minor changes to apply on the code. Great job!!

pablogrela committed:
Version 2.4.0

Manage push notifications
Renamed reserve to book
Other small changes

pablogrela committed:
Version 2.4.1

Update Push, Update activity if it is in the foreground

pablogrela committed:
Version 2.4.1

Corrections

pablogrela committed:
Version 2.5.0

Manage scheduler

pablogrela committed:
Version 2.5.1

Manage scheduler

pablogrela committed:
Version 2.5.2

Manage scheduler
Add viewpager by day in scheduler

pablogrela committed:
Version 2.5.3

Manage scheduler
Change urls and objects DTO

pablogrela committed:
Create google-services.json

ficiverson commented on app/build.gradle:
```diff
@@ -19,23 +19,32 @@
-        versionCode 1
-        versionName "1.0"
+        versionCode 7
+        versionName "2.5.3"
```
versionCode 2
versionName 1.1

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Book.java:
```diff
@@ -0,0 +1,261 @@
+                if (!intent.getBooleanExtra("notificationSkip", false)) {
+                    Intent i = new Intent(Book.this, PodcastingService.class);
+                    stopService(i);
+                    //notifyAdapterToRefresh();
```
delete unused code

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/CreateBook.java:
```diff
@@ -0,0 +1,301 @@
+        });
+
+        newDateStart = Calendar.getInstance();
+        dateStartPickerDialog = new DatePickerDialog(this, new DatePickerDialog.OnDateSetListener() {
```
keep code format

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/CreateReport.java:
```diff
@@ -0,0 +1,454 @@
+                e.printStackTrace();
+            }
+        } else if (requestCode == CAMERA_REQUEST) {
+            photo = (Bitmap) data.getExtras().get("data");
```
define"data" as public static final String on class

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/CreateReport.java:
```diff
@@ -0,0 +1,454 @@
+        if (requestCode == EXTERNAL_REQUEST || requestCode == CAMERA_REQUEST) {
+            photosMap.put(imageId, bitmapToByte(photo));
+
+            RelativeLayout view = new RelativeLayout(this);
```
Is necessary to update layout params on code? Why?

pablogrela replied:
To put button remove

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/CreateReport.java:
```diff
@@ -0,0 +1,454 @@
+        }
+
+        // To members, the values is evaluate in server
+        // report.setDescription(description.getHint() + description.getText().toString());
```
delete unused statments

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/FirebaseActivity.java:
```diff
@@ -1,8 +1,8 @@
 /*
  *
- *  * Copyright (C) 2016 @ Fernando Souto González
+ *  * Copyright (C) 2017 @ Pablo Grela
```
ResponseProgramDTO -> FirebaseActivty ? Check this please

pablogrela replied:
I remove ResponseProgramDTO, with RadioCo is not use.
Maybe create branch CUACFM of radiocom-android

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Home.java:
```diff
@@ -407,6 +484,44 @@ public void loadPodcast() {
+    public void loadScheduler() {
+        isSearchable = true;
+        invalidateOptionsMenu();
+//        TransmissionsPageFragment transmissionsPageFragment = new TransmissionsPageFragment();
```
delete unused statments

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Home.java:
```diff
@@ -474,17 +587,21 @@ public void onClick(DialogInterface dialog, int id) {
-            edit.putBoolean("isMediaPlaying", playing);
-            edit.commit();
+
+            new GetTransmissionNow(mContext, this).execute();
```
Rename gettransmissionnow -> getLiveBroadcast

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Home.java:
```diff
@@ -474,17 +587,21 @@ public void onClick(DialogInterface dialog, int id) {
+
+            new GetTransmissionNow(mContext, this).execute();
+
+//            audioIntent = new Intent(Home.this, StreamingService.class);
```
delete unused statements

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Home.java:
```diff
@@ -497,21 +614,110 @@ private void audioController() {
+        }
+    }
+
+    private Intent audioIntent;
```
declare this field on top of the class

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Home.java:
```diff
@@ -497,21 +614,110 @@ private void audioController() {
+
+    private Intent audioIntent;
+
+    public void setTransmissionDTO(TransmissionDTO transmissionDTO) {
```
Rename TransmissionDTO to LiveBroadcastDTO

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Home.java:
```diff
@@ -497,21 +614,110 @@ private void audioController() {
+                    .into(new Target() {
+                        @Override
+                        public void onBitmapLoaded(Bitmap bitmap, Picasso.LoadedFrom from) {
+                            ByteArrayOutputStream stream = new ByteArrayOutputStream();
```
What happend with a really big icon? please dowscale the bitmap to a max value

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Home.java:
```diff
@@ -497,21 +614,110 @@ private void audioController() {
+
+                        @Override
+                        public void onBitmapFailed(Drawable errorDrawable) {
+                            startService(audioIntent);
```
set the default icon in case bitmap load fails

pablogrela replied:
Already done in the other class

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Podcast.java:
```diff
@@ -52,30 +49,23 @@
- */
-public class Podcast extends AppCompatActivity{
+import static justforcommunity.radiocom.utils.GlobalValues.JSON_PODCAST;
+
```
plese do not delete created by

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Report.java:
```diff
@@ -0,0 +1,296 @@
+                if (!intent.getBooleanExtra("notificationSkip", false)) {
+                    Intent i = new Intent(Report.this, PodcastingService.class);
+                    stopService(i);
+                    //notifyAdapterToRefresh();
```
remove unused code

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Splash.java:
```diff
@@ -63,7 +54,7 @@ protected void onCreate(Bundle savedInstanceState) {
         getWindow().requestFeature(Window.FEATURE_ACTION_BAR);
         getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
-
+        Bundle b = getIntent().getExtras();
```
this is not necessary

ficiverson commented on app/src/main/java/justforcommunity/radiocom/activities/Splash.java:
```diff
@@ -73,70 +64,67 @@ protected void onCreate(Bundle savedInstanceState) {
-        GetStation stationTask = new GetStation(contex,activity);
-        stationTask.execute();
+    public void launchGetStations(Context contex, Splash activity) {
+        new GetStation(contex, activity).execute();
```
Please create an object GetStation an call execute as it was before

ficiverson commented on app/src/main/java/justforcommunity/radiocom/adapters/TransmissionListAdapter.java:
```diff
@@ -0,0 +1,170 @@
+
+import static justforcommunity.radiocom.utils.GlobalValues.radiocoURL;
+
+public class TransmissionListAdapter extends ArrayAdapter<TransmissionDTO> implements Filterable {
```
Rename to LiveBroadcastAdapter

ficiverson commented on app/src/main/java/justforcommunity/radiocom/fragments/TransmissionPagerFragment.java:
```diff
@@ -0,0 +1,164 @@
+import static justforcommunity.radiocom.utils.DateUtils.formatDate;
+
+
+public class TransmissionPagerFragment extends Fragment {
```
Rename to LiveBroadcast

ficiverson commented on app/src/main/java/justforcommunity/radiocom/fragments/TransmissionPagerFragment.java:
```diff
@@ -0,0 +1,164 @@
+    private ViewPager mViewPager;
+    private List<DayPage> mTabs = new ArrayList<>();
+
+    static class DayPage {
```
Do not create static inner class as the other viewpager on the app

ficiverson commented on app/src/main/java/justforcommunity/radiocom/fragments/TransmissionPagerFragment.java:
```diff
@@ -0,0 +1,164 @@
+    }
+
+    // Class to control number of page
+    class TransmissionPagerAdapter extends FragmentPagerAdapter {
```
Create adapter in other file not inner class

ficiverson replied:
Rename page adapter

ficiverson commented on app/src/main/java/justforcommunity/radiocom/fragments/TransmissionsPageFragment.java:
```diff
@@ -0,0 +1,149 @@
+
+import static justforcommunity.radiocom.utils.GlobalValues.JSON_PODCAST;
+
+public class TransmissionsPageFragment extends FilterFragment {
```
Rename class livebroadcast

ficiverson commented on app/src/main/java/justforcommunity/radiocom/model/ProgramDTO.java:
```diff
@@ -20,15 +20,33 @@
- */
 public class ProgramDTO {
 
+    private String id;
```
Please do not remove createdby

ficiverson commented on app/src/main/java/justforcommunity/radiocom/model/TransmissionDTO.java:
```diff
@@ -0,0 +1,111 @@
+
+import static justforcommunity.radiocom.utils.GlobalValues.baseURLWEB;
+
+public class TransmissionDTO {
```
Rename to LiveBroadcast

ficiverson commented on app/src/main/java/justforcommunity/radiocom/service/ServiceBase.java:
```diff
@@ -33,56 +33,54 @@
-/**
- * Created by iver on 7/3/16.
- */
+
```
Do not remove createdby

ficiverson commented on app/src/main/java/justforcommunity/radiocom/service/ServiceTransmissions.java:
```diff
@@ -0,0 +1,117 @@
+import static justforcommunity.radiocom.utils.GlobalValues.transmissionNowURL;
+import static justforcommunity.radiocom.utils.GlobalValues.transmissionsURL;
+
+
```
Rename this class

pablogrela replied:
Name?

ficiverson commented on app/src/main/java/justforcommunity/radiocom/service/ServiceTransmissions.java:
```diff
@@ -0,0 +1,117 @@
+
+            // Decode String, in some cases, it may not be necessary
+            decoded = response.getBody();
+            decoded = new String(response.getBody().getBytes("ISO-8859-1"));
```
why not UTF-8 the response must be always utf-8 to avoid enconding problems

pablogrela replied:
RadioCo not provides that encoding

ficiverson commented on app/src/main/java/justforcommunity/radiocom/service/ServiceTransmissions.java:
```diff
@@ -0,0 +1,117 @@
+            decoded = new String(response.getBody().getBytes("ISO-8859-1"));
+
+        } catch (RestClientException e) {
+            Log.d("ServiceTransmissions", "getTransmissionNow", e);
```
rename e to restClientException

ficiverson commented on app/src/main/java/justforcommunity/radiocom/service/ServiceTransmissions.java:
```diff
@@ -0,0 +1,117 @@
+    }
+
+    public List<TransmissionDTO> getTransmissions(String dateSearch) throws RestClientException, WebServiceStatusFailException {
+
```
delete unused code

ficiverson commented on app/src/main/java/justforcommunity/radiocom/service/ServiceTransmissions.java:
```diff
@@ -0,0 +1,117 @@
+//        endDate.add(Calendar.DAY_OF_MONTH, 0);
+//        String url = transmissionsURL + "?after=" + DateUtils.formatDate(startDate, DateUtils.FORMAT_DATE_GET) + "&before=" + DateUtils.formatDate(endDate, DateUtils.FORMAT_DATE_GET);
+
+        String url = transmissionsURL + "?after=" + dateSearch + "&before=" + dateSearch;
```
defime after and  before as constants

ficiverson commented on app/src/main/java/justforcommunity/radiocom/service/ServiceTransmissions.java:
```diff
@@ -0,0 +1,117 @@
+
+            // Decode String, in some cases, it may not be necessary
+            decoded = response.getBody();
+            decoded = new String(response.getBody().getBytes("ISO-8859-1"));
```
Why theres a lot of encodings? use utf-8

pablogrela replied:
RadioCo not provides that encoding

ficiverson commented on app/src/main/java/justforcommunity/radiocom/task/Book/SendAnswerBook.java:
```diff
@@ -0,0 +1,86 @@
+                book = serviceBooks.SendAnswerBook(bookId, answer, manage);
+                res = true;
+            } catch (RestClientException e) {
+                Log.d(TAG, "doInBackground()", e);
```
do not print everything to log

ficiverson commented on app/src/main/java/justforcommunity/radiocom/task/Book/SendBook.java:
```diff
@@ -0,0 +1,86 @@
+                message = "UserAlreadyBookException";
+                res = false;
+            } catch (RestClientException e) {
+                Log.d(TAG, "doInBackground()", e);
```
we dont need to print everything

ficiverson commented on app/src/main/java/justforcommunity/radiocom/task/Book/SendBook.java:
```diff
@@ -0,0 +1,86 @@
+                Log.d(TAG, "doInBackground()", e);
+                res = false;
+            } catch (Exception e) {
+                Log.d(TAG, "doInBackground()", e);
```
if you want to print in debug please set a debug mode to print

pablogrela replied:
i dont understand

ficiverson commented on app/src/main/java/justforcommunity/radiocom/task/DownloadEpisode.java:
```diff
@@ -22,139 +22,122 @@
             if (redirect) {
                 String newUrl = connection.getHeaderField("Location");
-                String cookies = connection.getHeaderField("Set-Cookie");
+                //String cookies = connection.getHeaderField("Set-Cookie");
```
provide explanation about this changes on this downloader

pablogrela replied:
I did not change anything, just format code

ficiverson commented on app/src/main/java/justforcommunity/radiocom/task/FirebaseUtils.java:
```diff
@@ -0,0 +1,86 @@
+                }
+            });
+            try {
+                countDownLatch.await(30L, TimeUnit.SECONDS);
```
this is not the best way. What about a slow 3G connection? provide another solution. Maybe you dont need a asynctask  is getToken a async method?

pablogrela replied:
getToken is a async method
I need token to authenticate
I dont know another solution

ficiverson commented on app/src/main/java/justforcommunity/radiocom/task/Report/SendAnswerReport.java:
```diff
@@ -0,0 +1,87 @@
+        if (cnn.isConnected(mContext)) {
+
+            try {
+                report = serviceReports.SendAnswerReport(reportId, answer, manage);
```
method must begin with lowercase

ficiverson commented on app/src/main/java/justforcommunity/radiocom/task/Transmissions/GetTransmissionNow.java:
```diff
@@ -0,0 +1,84 @@
+import justforcommunity.radiocom.utils.InternetConnection;
+
+
+public class GetTransmissionNow extends AsyncTask<Boolean, Float, Boolean> {
```
rename this class

ficiverson commented on app/src/main/java/justforcommunity/radiocom/task/Transmissions/GetTransmissions.java:
```diff
@@ -0,0 +1,77 @@
+import justforcommunity.radiocom.utils.InternetConnection;
+
+
+public class GetTransmissions extends AsyncTask<Boolean, Float, Boolean> {
```
Rename class

ficiverson commented on app/src/main/java/justforcommunity/radiocom/utils/MessagingService.java:
```diff
@@ -0,0 +1,163 @@
+            Log.d(TAG, "Message data payload: " + remoteMessage.getData());
+
+            Intent intent = null;
+            // String packageName = null;
```
unused code

ficiverson commented on app/src/main/java/justforcommunity/radiocom/utils/MessagingService.java:
```diff
@@ -0,0 +1,163 @@
+                    // Open book info
+                    intent = new Intent(getApplicationContext(), Book.class);
+                    intent.putExtra(JSON_BOOK, remoteMessage.getData().get("value"));
+                    //packageName = Book.class.getPackage().toString() + ".Book";
```
unused code

ficiverson commented on app/src/main/java/justforcommunity/radiocom/utils/MessagingService.java:
```diff
@@ -0,0 +1,163 @@
+                case REPORT:
+                    // Open report info
+                    intent = new Intent(getApplicationContext(), Report.class);
+                    intent.putExtra(JSON_REPORT, remoteMessage.getData().get("value"));
```
declare "value"final static string

ficiverson commented on app/src/main/java/justforcommunity/radiocom/utils/MessagingService.java:
```diff
@@ -0,0 +1,163 @@
+
+            // Create Notification,
+            PendingIntent localPendingIntent = PendingIntent.getActivity(this, 3, intent, PendingIntent.FLAG_UPDATE_CURRENT);
+            Bitmap myIcon = BitmapFactory.decodeResource(getResources(), R.drawable.ic_launcher);
```
check bitmap size and downscale. What happend with a very big image??

ficiverson commented on app/src/main/java/justforcommunity/radiocom/utils/MessagingService.java:
```diff
@@ -0,0 +1,163 @@
+        return false;
+    }
+
+//    private boolean isAppOnForeground(Context context, String packageName) {
```
remove unused code

ficiverson commented on app/src/main/java/justforcommunity/radiocom/utils/StreamingService.java:
```diff
@@ -92,7 +87,13 @@ private void showNotification(String paramString1, String paramString2) {
-        Bitmap myIcon = BitmapFactory.decodeResource(getResources(), R.drawable.ic_launcher);
+        Bitmap myIcon;
+        if (logo != null) {
+            myIcon = BitmapFactory.decodeByteArray(logo, 0, logo.length);
```
be careful with really big bitmaps

ficiverson commented on .idea/misc.xml:
```diff
@@ -27,36 +27,5 @@
-      </state>
-    </states>
-  </component>
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" assert-keyword="true" jdk-15="true" project-jdk-name="JDK" project-jdk-type="JavaSDK" />
```
remove your settings from PR

ficiverson commented on app/build.gradle:
```diff
@@ -19,23 +19,32 @@
+            storePassword '12345678'
+        }
+    }
+    compileSdkVersion 25
```
do not update compilesdkversion nor buildtoolsversion

pablogrela replied:
Fail compilation....

pablogrela committed:
Version 2.5.4

Improves

pablogrela committed:
Merge remote-tracking branch 'origin/master'

Commit:
Version 2.5.5
Format all files
Change the license presentation and copyright in all files
Library update

pablogrela committed:
2.5.6
Allow to deactivate the private part by means of a parameter that is received in the stationDTO
